src/frontend/src/optimizer/plan_node/logical_source.rs

```diff
@@ -302,6 +301,20 @@ impl LogicalSource {
             Ok(source.into())
         }
     }
+
+    fn is_kafka(&self) -> bool {
+        self.source_catalog()
+            .is_some_and(|s| s.with_properties.get(UPSTREAM_SOURCE_KEY).unwrap() == "kafka")
+    }
+
+    /// For kafka source, the column is always available. But when `CREATE TABLE` with kafka connector,
+    /// it's not available.
```

Comment:
the column in the source catalog, but we prune it when building a streaming job.
we do not know it because we cannot `describe source` 🤣